app/debug_overlay.conf

```diff
@@ -1 +1,11 @@
 CONFIG_DEBUG=y
+CONFIG_ASSERT=y
```

Comment:
Move `CONFIG_ASSERT=y` to MTL, TGL and TGL-H board files please.

Reply:
@aborisovich So I' mok with this, but  debug_overlay.conf already enables it for all platforms in the SOF CI, so let's get this enabled first. If CI is green, we can move to product definitions as well.